### ISSUE TITLE
NVDA fails to correctly announce deleted text when using Control+Backspace

### ISSUE THREAD

**Issue by tarikhadzirovicofficial** [nvaccess/nvda#17630]
NVDA incorrectly announces parts of words when the `Control+Backspace` shortcut is used to delete text. Instead of announcing only the deleted word, NVDA announces the entire word preceding the cursor, even if only part of it was deleted.

#### Steps to reproduce:

1. Open any edit field.
2. Type "NonVisual Desktop Access" as an example.
3. Place the cursor between the letters "n" and "V".
4. Press `Control+Backspace`.

#### Actual behavior:

NVDA announces "NonVisual," even though only "Non" (the word preceding the cursor) is deleted.

#### Expected behavior:

NVDA should announce only "Non," since that is the deleted part of the word "NonVisual".

#### Additional information:

- **NVDA version:** NVDA 2024.4 34423 (installed).
- **Windows version:** Windows 11 23H2 (AMD64) build 22631.4602.
- **Software:** Issue occurs in any edit field.

#### Does the issue still occur after restarting your computer?

Yes.

#### Have you tried any other versions of NVDA? If so, please report their behaviors.

Yes, the behaviors are the same.

#### If NVDA add-ons are disabled, is your problem still occurring?

Yes.

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?

Yes.

**Comment by WxnChen11** [comment]
Im interested in working on this, does anyone have insights on where to begin?

**Comment by SaschaCowley** [comment]
@WxnChen11 I'd try starting with https://github.com/nvaccess/nvda/blob/19296bf4fde5e3f78150fcd04075ebcc841c3cdd/source/editableText.py#L267-L295
